hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -919,24 +919,37 @@ class EmitMethodBuilder[C](
     }
   }
 
+  private def _invoke[T](_args: Param*): Code[T] = {
+    EmitCodeBuilder.scopedCode(this) { cb =>
+      val args = _args.map {
+        case cp: CodeParam =>
+          cp
+        case EmitParam(EmitCode(setup, m, pv)) =>
+          cb += setup
+          if (pv.pt.required) {
+            cb += Code.toUnit(m) // side effects???, is this even possible
```

Comment:
Yes, it is definitely possible.  In particular, an arbitrary amount of setup can be put into m, and in particular

EmitCode(setup, m, pv) == EmitCode(Code._empty, Code(setup, m), pv)